pandas/tests/series/test_ufunc.py

```diff
@@ -0,0 +1,185 @@
+    a2[::4] = 0
+    return a1, a2
+
+
+@pytest.mark.parametrize("ufunc", UNARY_UFUNCS)
+@pytest.mark.parametrize("sparse", SPARSE, ids=SPARSE_IDS)
+def test_unary_ufunc(ufunc, sparse):
+    array = np.random.randint(0, 10, 10, dtype='int64')
+    array[::2] = 0
+    if sparse:
+        array = pd.SparseArray(array, dtype=pd.SparseDtype('int', 0))
+
+    index = list(string.ascii_letters[:10])
+    name = "name"
+    series = pd.Series(array, index=index, name=name)
+
+    result = ufunc(series)
+    expected = pd.Series(ufunc(array), index=index, name=name)
+    tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("ufunc", BINARY_UFUNCS)
+@pytest.mark.parametrize("sparse", SPARSE, ids=SPARSE_IDS)
+@pytest.mark.parametrize("shuffle", SHUFFLE)
+@pytest.mark.parametrize("box_other", ['series', 'index', 'raw'])
+@pytest.mark.parametrize("flip", [True, False],
+                         ids=['flipped', 'straight'])
+def test_binary_ufunc(ufunc, sparse, shuffle, box_other,
+                      flip,
+                      arrays_for_binary_ufunc):
+    # Check the invariant that
+    #   ufunc(Series(a), Series(b)) == Series(ufunc(a, b))
+    #   with alignment.
+    a1, a2 = arrays_for_binary_ufunc
+    if sparse:
+        a1 = pd.SparseArray(a1, dtype=pd.SparseDtype('int', 0))
+        a2 = pd.SparseArray(a2, dtype=pd.SparseDtype('int', 0))
+
+    name = "name"
+    s1 = pd.Series(a1, name=name)
+    if box_other == 'series':
+        s2 = pd.Series(a2, name=name)
+    elif box_other == 'index':
+        # Index should defer to Series
+        s2 = pd.Index(a2, naame=name)
+    else:
+        s2 = a2
+
+    idx = np.random.permutation(len(s1))
+
+    if shuffle:
+        s2 = s2.take(idx)
+        if box_other != 'series':
+            # when other is a Series, we align, so we don't
+            # need to shuffle the array for expected. In all
+            # other cases, we do.
+            a2 = a2.take(idx)
+
+    a, b = s1, s2
+    c, d = a1, a2
+
+    if flip:
+        a, b = b, a
+        c, d = d, c
+
+    result = ufunc(a, b)
+    expected = pd.Series(ufunc(c, d), name=name)
+    if box_other == 'index' and flip:
+        raise pytest.xfail("Index should defer to Series")
+    tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("ufunc", BINARY_UFUNCS)
+@pytest.mark.parametrize("sparse", SPARSE, ids=SPARSE_IDS)
+@pytest.mark.parametrize("flip", [True, False])
+def test_binary_ufunc_scalar(ufunc, sparse, flip, arrays_for_binary_ufunc):
+    array, _ = arrays_for_binary_ufunc
+    if sparse:
+        array = pd.SparseArray(array)
+    other = 2
+    series = pd.Series(array, name="name")
+
+    a, b = series, other
+    c, d = array, other
+    if flip:
+        c, d = b, c
+        a, b = b, a
+
+    expected = pd.Series(ufunc(a, b), name="name")
+    result = pd.Series(ufunc(c, d), name="name")
+    tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("ufunc", [np.divmod])  # any others?
```

Comment:
np.modf (what you just added)

Reply:
modf is a unary function with two outputs. This test is for binary functions with one output.